### ISSUE TITLE
The partial view  was not found or no view engine supports the searched locations

### ISSUE THREAD

**Issue by khosro** [RazorGenerator/RazorGenerator#22]
Hi,
I modified https://github.com/davidebbo/MvcApplicationRazorGeneratorSeparateLibrary project as follows :
1.Add a class named TestHelper.cs to "LibraryViewProject" and source code is :

		using System;
		using System.Collections.Generic;
		using System.Linq;
		using System.Web;
		using System.Web.Mvc;
		using System.Web.Mvc.Html;
		namespace LibraryViewProject
		{
			public static class TestHelper
			{
				public static MvcHtmlString Confirm(this HtmlHelper html)
				{
				   
					return html.Partial( @"~\Views\Shared\Test\Confirm.cshtml");
				}
			}
		}

And then in "LibraryViewProject" project, i create a cshtml file "Views\Shared\Test\Confirm.cshtml"  ,source code is :

      @{
	     Layout = null;
	  }

	<h1>Confirm Test</h1>

2.Remove "Views/Home/Index.cshtml" from "LibraryViewProject " and add it to "MvcApplicationRazorGeneratorSeparateLibrary"  and add the following line to it :

     @Html.Confirm();

After running project the the following excetion occurred:

		The partial view '~\Views\Shared\Test\Confirm.cshtml' was not found or no view engine supports the searched locations. The following locations were searched:
		~\Views\Shared\Test\Confirm.cshtml

		Description: An unhandled exception occurred during the execution of the current web request. Please review the stack trace for more information about the error and where it originated in the code. 

		Exception Details: System.InvalidOperationException: The partial view '~\Views\Shared\Test\Confirm.cshtml' was not found or no view engine supports the searched locations. The following locations were searched:
		~\Views\Shared\Test\Confirm.cshtml

		Source Error: 

		Line 13:         {
		Line 14:            
		Line 15:             return html.Partial( @"~\Views\Shared\Test\Confirm.cshtml");
		Line 16:         }
		Line 17:     }

How to solve this exception?

**Comment by khosro** [comment]
I think this issue is related to following code in PrecompiledMvcEngine.cs

      base.PartialViewLocationFormats = new[] {
                "~/Views/{1}/{0}.cshtml", 
                "~/Views/Shared/{0}.cshtml", 
            };

After adding `"~/Views/Shared/Test/{0}.cshtml"` to `base.PartialViewLocationFormats ` the exception  solved.

**Comment by davidebbo** [comment]
Ok, but 'test' in an arbitrary token here, right? If so, it probably doesn't make sense to add that line for everyone. Maybe there can be alternate extensibility in the engine to support this.

**Comment by khosro** [comment]
Yes,it is arbitrary token.But i choose `Test` because ,i added a "Helper" folder to `~/Views/Shared` and add my own helper views in it.But it seems RazorGenerator does not generate `generated.cs` file under the *.cshtml files.So that i changed it to `Test`.And also the same issue exists for any folder which has `Helper` in its name.

**Comment by akonyer** [comment]
Just noticed this happen in our project as well. Any view with "Helper" in the name did not get compiled.
Changing the view name worked, but was definitely a lot of time wasted trying to figure out what was going on.

**Comment by HenryQW** [comment]
@akonyer 
Ran into the same thing, I can't believe it's been 2 years and it's still not fixed.
